dataclients/kubernetes/routegroup.go

```diff
@@ -0,0 +1,554 @@
+	if s.Spec.ClusterIP == "" {
+		return "", errMissingClusterIP
+	}
+
+	return fmt.Sprintf("http://%s:%d", s.Spec.ClusterIP, backend.ServicePort), nil
+}
+
+func applyServiceBackend(ctx *routeGroupContext, backend *skipperBackend, r *eskip.Route) error {
+	s, err := getBackendService(ctx, backend)
+	if err != nil {
+		return err
+	}
+
+	targetPort, ok := s.getTargetPortByValue(backend.ServicePort)
+	if !ok {
+		return targetPortNotFound(backend.ServiceName, backend.ServicePort)
+	}
+
+	eps := ctx.clusterState.getEndpointsByTarget(
+		namespaceString(ctx.routeGroup.Metadata.Namespace),
+		s.Meta.Name,
+		targetPort,
+	)
+
+	if len(eps) == 0 {
+		b, err := createClusterIPBackend(s, backend)
+		if err != nil {
+			return err
+		}
+
+		log.Infof(
+			"[routegroup] Target endpoints not found, using service cluster IP as a fallback for %s/%s %s:%d",
+			namespaceString(ctx.routeGroup.Metadata.Namespace),
+			ctx.routeGroup.Metadata.Name,
+			backend.ServiceName,
+			backend.ServicePort,
+		)
+
+		r.BackendType = eskip.NetworkBackend
+		r.Backend = b
+		return nil
+	}
+
+	if len(eps) == 1 {
+		r.BackendType = eskip.NetworkBackend
+		r.Backend = eps[0]
+		return nil
+	}
+
+	r.BackendType = eskip.LBBackend
+	r.LBEndpoints = eps
+	r.LBAlgorithm = defaultLoadBalancerAlgorithm
+	if backend.Algorithm != loadbalancer.None {
+		r.LBAlgorithm = backend.Algorithm.String()
+	}
+
+	return nil
+}
+
+func applyDefaultFilters(ctx *routeGroupContext, serviceName string, r *eskip.Route) error {
+	f, err := ctx.defaultFilters.getNamed(namespaceString(ctx.routeGroup.Metadata.Namespace), serviceName)
+	if err != nil {
+		return defaultFiltersError(ctx.routeGroup.Metadata, serviceName, err)
+	}
+
+	// safe to prepend as defaultFilters.get() copies the slice:
+	r.Filters = append(f, r.Filters...)
+	return nil
+}
+
+func applyBackend(ctx *routeGroupContext, backend *skipperBackend, r *eskip.Route) error {
+	r.BackendType = backend.Type
+	switch r.BackendType {
+	case serviceBackend:
+		if err := applyServiceBackend(ctx, backend, r); err != nil {
+			return err
+		}
+	case eskip.NetworkBackend:
+		r.Backend = backend.Address
+	case eskip.LBBackend:
+		r.LBEndpoints = backend.Endpoints
+		r.LBAlgorithm = defaultLoadBalancerAlgorithm
+		r.LBAlgorithm = backend.Algorithm.String()
+	}
+
+	return nil
+}
+
+func appendPredicate(p []*eskip.Predicate, name string, args ...interface{}) []*eskip.Predicate {
+	return append(p, &eskip.Predicate{
+		Name: name,
+		Args: args,
+	})
+}
+
+func storeHostRoute(ctx *routeGroupContext, r *eskip.Route) {
+	for _, h := range ctx.hosts {
+		ctx.hostRoutes[h] = append(ctx.hostRoutes[h], r)
+	}
+}
+
+func appendEastWest(ctx *routeGroupContext, routes []*eskip.Route, current *eskip.Route) []*eskip.Route {
+	// how will the route group name for the domain name play together with
+	// zalando.org/v1/stackset and zalando.org/v1/fabricgateway? Wouldn't it be better to
+	// use the service name instead?
```

Comment:
It should just work with stackset and and fabricgateway. We don't depend on versioned stacks in the traffic switching case, so we also do not have this in RouteGroup nor in the east-west case.